lms/djangoapps/django_comment_client/forum/views.py

```diff
@@ -124,6 +128,16 @@ def get_threads(request, course_key, discussion_id=None, per_page=THREADS_PER_PA
 
     threads, page, num_pages, corrected_text = cc.Thread.search(query_params)
 
+    # If not provided with a discussion id, filter threads by commentable ids
+    # which are accessible to the current user.
+    if discussion_id is None:
+        commentable_ids = (
+            course.top_level_discussion_topic_ids + utils.get_discussion_id_map(course, request.user).keys()
```

Comment:
instead of `get_discussion_id_map`, could we use `get_accessible_discussion_modules` instead?  it seems like that will have the same data, but skip a bunch of transformation that won't get used anyway.